attestation-service/verifier/src/se/ibmse.rs

```diff
@@ -0,0 +1,318 @@
+                warn!("The host key document in '{hkd}' contains more than one certificate!")
+            }
+            // Panic: len is == 1 -> unwrap will succeed/not panic
+            let c = certs.first().unwrap();
+            verifier.verify(c)?;
+            arcb.add_hostkey(c.public_key()?);
+        }
+        let encr_ctx = ReqEncrCtx::random(SymKeyType::Aes256)?;
+        let request_blob = arcb.encrypt(&encr_ctx)?;
+        
+        let mut file = File::open(pub_key_file)?;
+        let mut contents = vec![];
+        file.read_to_end(&mut contents)?;
+        let rsa = Rsa::public_key_from_pem(&contents)?;
+        let rsa_public_key = &PKey::from_rsa(rsa)?;
+
+        let conf_data = arcb.confidential_data();
+        let encr_measurement_key =
+            encrypt_measurement_key(conf_data.measurement_key(), rsa_public_key);
+        let encr_request_nonce =
+            encrypt_measurement_key(conf_data.nonce().clone().unwrap().value(), rsa_public_key);
+
+        Ok(Self {
+            request_blob,
+            measurement_size: meas_alg.exp_size(),
+            additional_size: arcb.flags().expected_additional_size(),
+            encr_measurement_key,
+            encr_request_nonce,
+            image_hdr_tags: *image_hdr_tags,
+        })
+    }
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SeAttestationResponse {
+    #[serde_as(as = "Base64")]
+    measurement: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    additional_data: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    user_data: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    cuid: ConfigUid,
+    #[serde_as(as = "Base64")]
+    encr_measurement_key: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    encr_request_nonce: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    image_hdr_tags: BootHdrTags,
+}
+
+impl SeAttestationResponse {
+    pub fn from_slice(response: &[u8]) -> Result<Self> {
+        Ok(serde_json::from_slice(response).unwrap())
+    }
+
+    pub fn from_string(request: &str) -> Result<Self> {
+        Ok(serde_json::from_str(request).unwrap())
+    }
+
+    pub fn create(
+        measurement: &[u8],
+        additional_data: &[u8],
+        user_data: &[u8],
+        cuid: &ConfigUid,
+        encr_measurement_key: &[u8],
+        encr_request_nonce: &[u8],
+        image_hdr_tags: &BootHdrTags,
+    ) -> Result<Self> {
+        Ok(Self {
+            measurement: measurement.to_vec(),
+            additional_data: additional_data.to_vec(),
+            user_data: user_data.to_vec(),
+            cuid: *cuid,
+            encr_measurement_key: encr_measurement_key.to_vec(),
+            encr_request_nonce: encr_request_nonce.to_vec(),
+            image_hdr_tags: *image_hdr_tags,
+        })
+    }
+
+    pub fn verify(&self, priv_key_file: &str) -> Result<SeAttestationClaims> {
+        let mut file = File::open(priv_key_file)?;
+        let mut contents = vec![];
+        file.read_to_end(&mut contents)?;
```

Comment:
Does it make sense to re-read and re-process the file in every verify operation or could the private key be initialized once?

background: the top-level fns in the verifiers are async and we shouldn't perform costly sync code in those. There are async fs operations, but maybe we don't need it if we parse the key once.

Reply:
OK, I'll use lazy load and cache in memory to do that. Thanks!